microsoft/surface/repos.nix

```diff
@@ -1,9 +1,7 @@
 { lib, pkgs, fetchFromGitHub, fetchurl }: {
-  linux-surface = fetchFromGitHub {
-    owner = "linux-surface";
-    repo = "linux-surface";
-    rev = "4130746c31606aebaccef58997b650beda9049f7";
-    sha256 = "0r74z0hj3rg8bn2arh2zj8p7awk7fi14gpyrs5h44lwxw4351xky";
+  linux-surface-kernel = fetchurl {
+    url = "https://github.com/linux-surface/kernel/archive/v5.16-surface.tar.gz";
```

Comment:
This is a branch, we should only reference commits or tags here. Otherwise the checksum will break.

Reply:
Also I don't quite understand the switch away from `fetchFromGitHub` here.

Reply:
That's my bad; I misunderstood what `fetchFromGitHub` did and accidentally thought it didn't apply in this situation (I'm still a beginner, and nix isn't easy to pick up).
I'll fix it so it points to a specific revision too.